### ISSUE TITLE
docker prune Error connecting: Error while fetching server API version: Not supported URL scheme http+docker

### ISSUE THREAD

**Issue by Reedler01** [ansible-collections/community.docker#938]
##### SUMMARY

Running the latest version of centos 9 stream. The community.docker.docker_prune module is returning Error connecting: Error while fetching server API version: Not supported URL scheme http+docker 
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

name: Prune everything
      community.docker.docker_prune:
        containers: true
        images: true
        images_filters:
          dangling: true
        networks: true
        volumes: true
        builder_cache: true
      register: docker_prune_output
##### ANSIBLE VERSION

```paste below
ansible [core 2.16.1]
```

##### COLLECTION VERSION

```paste below
#16 0.727 community.aws                 6.4.0  
#16 0.727 community.azure               2.0.0  
#16 0.727 community.ciscosmb            1.0.7  
#16 0.727 community.crypto              2.16.0 
#16 0.727 community.digitalocean        1.24.0 
#16 0.727 community.dns                 2.6.3  
#16 0.727 community.docker              3.4.10 
#16 0.727 community.fortios             1.0.0  
#16 0.727 community.general             7.5.1  
#16 0.727 community.google              1.0.0  
#16 0.727 community.grafana             1.6.1  
#16 0.727 community.hashi_vault         5.0.1  
#16 0.727 community.hrobot              1.8.1  
#16 0.727 community.libvirt             1.3.0  
#16 0.727 community.mongodb             1.6.3  
#16 0.727 community.mysql               3.8.0  
#16 0.727 community.network             5.0.2  
#16 0.727 community.okd                 2.3.0  
#16 0.727 community.postgresql          2.4.3  
#16 0.727 community.proxysql            1.5.1  
#16 0.727 community.rabbitmq            1.2.3  
#16 0.727 community.routeros            2.10.0 
#16 0.727 community.sap                 1.0.0  
#16 0.727 community.sap_libs            1.4.1  
#16 0.727 community.skydive             1.0.0  
#16 0.727 community.sops                1.6.7  
#16 0.727 community.vmware              3.11.1 
#16 0.727 community.windows             1.13.0 
#16 0.727 community.zabbix              2.1.0 
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = None
```

##### OS / ENVIRONMENT

Ansible running on alpine:3.19.2
python version = 3.11.9
This is part of the basehttpadapter.py file 
    # Fix for requests 2.32.2+:
    # https://github.com/psf/requests/commit/c98e4d133ef29c46a9b68cd783087218a8075e05
    def get_connection_with_tls_context(self, request, verify, proxies=None, cert=None):
        return self.get_connection(request.url, proxies)
##### STEPS TO REPRODUCE

Run the above playbook, this is working on older centos versions and ubuntu hosts. 

```yaml
    - name: Run Docker Compose
      ansible.builtin.command: docker-compose -f "{{ item }}" up -d
      loop:
        - "/home/$USER/runner1/runner1-compose.yml"
        - "/home/$USER/runner2/runner2-compose.yml"

    - name: Prune everything
      community.docker.docker_prune:
        containers: true
        images: true
        images_filters:
          dangling: true
        networks: true
        volumes: true
        builder_cache: true
      register: docker_prune_output

    - name: Show Reclaimed Disk Space
      ansible.builtin.debug:
        var: docker_prune_output

    - name: DNF update centos
      ansible.builtin.yum:
        name: "*"
        state: latest
      register: result

    - name: Show output
      ansible.builtin.debug:
        msg: "{{ result.results }}"

    - name: Enable service firewalld and ensure it is not masked
      ansible.builtin.systemd:
        name: firewalld
        enabled: true
        masked: false
        state: started

    - name: SELinux in permissive mode, logging actions that would be blocked.
      ansible.posix.selinux:
        policy: targeted
        state: permissive

    - name: Check if reboot is required
      ansible.builtin.shell: needs-restarting -r
      failed_when: false
      register: reboot_required
      changed_when: false

    - name: Reboot
      ansible.builtin.reboot:
      when: reboot_required.rc != 0
```

##### EXPECTED RESULTS

This will not error on docker prune

##### ACTUAL RESULTS

Receiving the error on this module Error connecting: Error while fetching server API version: Not supported URL scheme http+docker

```paste below

```

**Comment by felixfontein** [comment]
You are using a very outdated version of community.docker (3.4.10). This issue has been fixed in [version 3.10.2](https://github.com/ansible-collections/community.docker/blob/main/CHANGELOG.rst#v3102). Please upgrade.

**Comment by Reedler01** [comment]
Thank you that fixed the issue. Sorry I missed that one!
